packages/core/src/Table/stories/TableSection/TableEditable.tsx

```diff
@@ -401,7 +438,9 @@ const Table = <T extends Data>({
       event.preventDefault();
       const formData = new FormData(event.currentTarget);
       const asset = Object.fromEntries(formData.entries());
-      await onRowUpdate?.({ id, ...asset } as T);
+      const formState = editRows.find((r) => r.id === id);
+      const mergedFormData = { ...asset, ...formState };
+      await onRowUpdate?.({ id, ...mergedFormData } as T);
```

Comment:
I left this because I'm having trouble using the switch. it seems to only register the switch value when it goes from unchecked to checked. So if I were to edit the second row status to `close` it does register the change

Reply:
If we remove the `value` property and replace `defaultValue` to:

```tsx
  defaultChecked={edit ? row.original[String(cols[1].accessor)]?.toString() === "Open" : false}
```

I think it helps. Then, on save and edit, we can look at the `status` value to add/edit the row: if it's equal to `on` the status is "Open"; otherwise it's "Closed". What do you think? I'm I missing something?

Reply:
Yes, you are wright it works, thanks! I updated it now.